projects/rocprofiler-systems/tests/rocprof-sys-gpu-connect-tests.cmake

```
@@ -0,0 +1,93 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# -------------------------------------------------------------------------------------- #
+#
+# GPU connectivity tests (transferBench)
+#
+# -------------------------------------------------------------------------------------- #
+
+set(_gpu_connect_environment
+    "ROCPROFSYS_ROCM_DOMAINS=hip_runtime_api,memory_copy,hsa_api"
+    "ROCPROFSYS_AMD_SMI_METRICS=busy,temp,power,xgmi,pcie"
+    "ROCPROFSYS_SAMPLING_CPUS=none"
+)
+
+set(_gpu_connect_rocpd_validation_rules
+    "${CMAKE_CURRENT_LIST_DIR}/rocpd-validation-rules/gpu-connect/validation-rules.json"
+    "${CMAKE_CURRENT_LIST_DIR}/rocpd-validation-rules/gpu-connect/amd-smi-rules.json"
+)
+
+# Enable ROCPD for tests only if valid ROCm is installed and a valid GPU is detected
+if(${ENABLE_ROCPD_TEST} AND ${_VALID_GPU})
+    list(APPEND _gpu_connect_environment "ROCPROFSYS_USE_ROCPD=ON")
+endif()
+
+set(skip_validation FALSE)
+
+if(EXISTS "${PROJECT_BINARY_DIR}/transferBench")
+    execute_process(
+        COMMAND ${PROJECT_BINARY_DIR}/transferBench
+        OUTPUT_VARIABLE _transfer_output
+        ERROR_VARIABLE _transfer_output
+        RESULT_VARIABLE _transfer_result
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_STRIP_TRAILING_WHITESPACE
+    )
+
+    if(_transfer_output MATCHES "Error: No valid transfers created")
+        set(skip_validation TRUE)
+    endif()
+endif()
+
+rocprofiler_systems_add_test(
+    SKIP_BASELINE SKIP_RUNTIME SKIP_REWRITE
+    NAME transferbench
+    TARGET transferBench
+    GPU ON
+    ENVIRONMENT "${_base_environment};${_gpu_connect_environment}"
+    LABELS "transferbench"
+    SAMPLING_SKIP_REGEX "Error: No valid transfers created"
+)
+
+if(NOT skip_validation)
+    rocprofiler_systems_add_validation_test(
+        NAME transferbench-sampling
+        PERFETTO_FILE "perfetto-trace.proto"
+        LABELS "transferbench;perfetto"
+        ARGS --counter-names "XGMI Read Data" "XGMI Write Data" -p
+    )
+
+    if(NOT skip_validation AND ${ENABLE_ROCPD_TEST} AND ${_VALID_GPU})
```

```suggestion
    if(${ENABLE_ROCPD_TEST} AND ${_VALID_GPU})
```
The "NOT skip_validation" check was redundant.

---

Thanks David. There was also a CI failure on Ubuntu 24.04 • gfx950 due to a CI timeout:

2025-11-11T03:10:51.0065833 [rocprof-sys][145773][63][ensure_ci_timeout_backtrace] timeout after  **120.000 seconds**... Generating backtraces for 14 threads...
2025-11-11T03:10:51.0067158Z Backtrace:

I am turning off sampling by default to see if that can bypass the timeout error